build-logic/src/main/kotlin/dokkabuild/tasks/GitCheckoutTask.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package dokkabuild.tasks
+
+import org.eclipse.jgit.api.Git
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileSystemOperations
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import javax.inject.Inject
+
+/**
+ * Clones a remote Git repository into [GitCheckoutTask.destination].
+ */
+@CacheableTask
+abstract class GitCheckoutTask @Inject constructor(
+    private val fs: FileSystemOperations
+) : DefaultTask() {
+
+    @get:OutputDirectory
+    abstract val destination: DirectoryProperty
+
+    /** Public HTTP URI of the Git repo. */
+    @get:Input
+    abstract val uri: Property<String>
+
+    /** Specific git hash to checkout. */
+    @get:Input
+    abstract val commitId: Property<String>
+
+    init {
+        group = "git checkout"
+    }
+
+    @TaskAction
+    fun action() {
+        val uri = uri.get()
+        val commitId = commitId.get()
+
+        fs.delete { delete(temporaryDir) }
+
+        Git.cloneRepository()
+            .setURI(uri)
+            .setDirectory(temporaryDir)
+            .call().use { git ->
+                git.pull().call()
+
+                git.checkout()
+                    .setName(commitId)
+                    .call()
+
+                fs.sync {
```

Comment:
minor: may be we can move `fs.sync` outside of `use { git -> ... }` block to better understand steps here?

Reply:
sure, makes sense to me